### PR TITLE
Add Google Cloud Trace API plugin

### PR DESCRIPTION
**Pull request by mkinoshi**
Hello, I am adding a plugin that connects your application with Stackdriver Trace API for Fastify. It measures how long each event takes within one request and generates trace results. FYI, it doesn't work with node 6, so I am trying to merge to `next`. It works on the current Fasitfy version as long as the node version is 8 or higher though. Hopefully, it is helpful for some people.

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: docs/Ecosystem.md [hunk 1]
```diff
@@ -79,6 +79,7 @@ Plugins maintained by the fastify team are listed under [Core](#core) while plug
 - [`fastify-google-cloud-storage`](https://github.com/carlozamagni/fastify-google-cloud-storage) Fastify plugin that exposes a GCP Cloud Storage client instance.
 - [`fastify-gql`](https://github.com/mcollina/fastify-gql) A GraphQL server implementation for Fastify with caching and [`graphql-jit`](https://github.com/ruiaraujo/graphql-jit).
 - [`fastify-graceful-shutdown`](https://github.com/hemerajs/fastify-graceful-shutdown) Shutdown Fastify gracefully and asynchronously.
+- [`fastify-gcloud-trace`](https://https://github.com/mkinoshi/fastify-gcloud-trace) Google Cloud Trace API Connector for Fastify.
```
Comment (jsumners) [inline docs/Ecosystem.md:82]: This should be in alphabetical order. So, it should be `fastify-gcloud-trace` followed by `fastify-google-cloud-storage`.

Comment (mkinoshi) [reply docs/Ecosystem.md]: Oops my bad, I updated it

File: docs/Ecosystem.md [hunk 2]
```diff
@@ -77,6 +77,7 @@ Plugins maintained by the fastify team are listed under [Core](#core) while plug
 - [`fastify-firebase-auth`](https://github.com/oxsav/fastify-firebase-auth) Firebase Authentication for Fastify supporting all of the methods relating to the authentication API.
 - [`fastify-firestore`](https://github.com/now-ims/fastify-firestore) Fastify plugin for [Google Cloud Firestore](https://cloud.google.com/nodejs/docs/reference/firestore/0.13.x/).
 - [`fastify-google-cloud-storage`](https://github.com/carlozamagni/fastify-google-cloud-storage) Fastify plugin that exposes a GCP Cloud Storage client instance.
+- [`fastify-gcloud-trace`](https://https://github.com/mkinoshi/fastify-gcloud-trace) [Google Cloud Trace API](https://cloud.google.com/trace/docs/reference) Connector for Fastify.
```
Comment (jsumners) [inline docs/Ecosystem.md:80]: "gcloud" alphabetizes higher in priority than "google".

Comment (mkinoshi) [reply docs/Ecosystem.md:80]: Omg you're right :/

Comment (mkinoshi) [reply docs/Ecosystem.md:79]: should be good now

Comment (mkinoshi) [reply docs/Ecosystem.md:79]: Sorry one sec

File: docs/Ecosystem.md [hunk 3]
```diff
@@ -72,11 +72,11 @@ Plugins maintained by the fastify team are listed under [Core](#core) while plug
 - [`fastify-explorer`](https://github.com/Eomm/fastify-explorer) Get control of your decorators across all the encapsulated contexts
 - [`fastify-esso`](https://github.com/patrickpissurno/fastify-esso) The easiest authentication plugin for Fastify, with built-in support for Single sign-on (and great documentation)
 - [`fastify-favicon`](https://github.com/smartiniOnGitHub/fastify-favicon) Fastify plugin to serve default favicon.
+- [`fastify-gcloud-trace`](https://https://github.com/mkinoshi/fastify-gcloud-trace) [Google Cloud Trace API](https://cloud.google.com/trace/docs/reference) Connector for Fastify.
```
Comment (jsumners) [inline docs/Ecosystem.md]: The previous commit was correct. The order is:

- `fastify-firestore`
- `fastify-gcloud-trace`
- `fastify-google-cloud-storage`

Comment (mkinoshi) [reply docs/Ecosystem.md]: Yeah the last one fixed it

File: docs/Ecosystem.md [hunk 4]
```diff
@@ -76,6 +76,7 @@ Plugins maintained by the fastify team are listed under [Core](#core) while plug
 - [`fastify-file-upload`](https://github.com/huangang/fastify-file-upload) Fastify plugin for uploading files.
 - [`fastify-firebase-auth`](https://github.com/oxsav/fastify-firebase-auth) Firebase Authentication for Fastify supporting all of the methods relating to the authentication API.
 - [`fastify-firestore`](https://github.com/now-ims/fastify-firestore) Fastify plugin for [Google Cloud Firestore](https://cloud.google.com/nodejs/docs/reference/firestore/0.13.x/).
+- [`fastify-gcloud-trace`](https://https://github.com/mkinoshi/fastify-gcloud-trace) [Google Cloud Trace API](https://cloud.google.com/trace/docs/reference) Connector for Fastify.
```
Comment (Eomm) [inline docs/Ecosystem.md:79]:
```suggestion
- [`fastify-gcloud-trace`](https://github.com/mkinoshi/fastify-gcloud-trace) [Google Cloud Trace API](https://cloud.google.com/trace/docs/reference) Connector for Fastify.
```

**Comment by mkinoshi** [comment]
@mcollina I'll close this PR now, and will create a new PR to `master` in that case!

Closing this PR to merge to master instead of next branch

**Comment by jsumners** [comment]
Could have just changed the target of this PR.

**Pull request closed without merging**